Core/MIPS/IR/IRCompFPU.cpp

```diff
@@ -0,0 +1,246 @@
+	{
+		ir.Write(IROp::LoadFloat, ft, rs, ir.AddConstant(offset));
+	}
+	break;
+	case 57: //Memory::Write_U32(FI(ft), addr); break; //swc1
+	{
+		ir.Write(IROp::StoreFloat, ft, rs, ir.AddConstant(offset));
+	}
+	break;
+
+	default:
+		_dbg_assert_msg_(CPU, 0, "Trying to interpret FPULS instruction that can't be interpreted");
+		break;
+	}
+}
+
+void IRJit::Comp_FPUComp(MIPSOpcode op) {
+	DISABLE;
+
+	int opc = op & 0xF;
+	if (opc >= 8) opc -= 8; // alias
+	if (opc == 0) {  // f, sf (signalling false)
+		ir.Write(IROp::ZeroFpCond);
+		return;
+	}
+
+	int fs = _FS;
+	int ft = _FT;
+
+	IROp irOp;
+	switch (opc) {
+	case 1:      // un,  ngle (unordered)
+		irOp = IROp::FCmpUnordered;
+		break;
+	case 2:      // eq,  seq (equal, ordered)
+		irOp = IROp::FCmpEqual;
+		break;
+	case 3:      // ueq, ngl (equal, unordered)
+		irOp = IROp::FCmpEqualUnordered;
+		return;
+	case 4:      // olt, lt (less than, ordered)
+		irOp = IROp::FCmpLessOrdered;
+		break;
+	case 5:      // ult, nge (less than, unordered)
+		irOp = IROp::FCmpLessUnordered;
+		break;
+	case 6:      // ole, le (less equal, ordered)
+		irOp = IROp::FCmpLessEqualOrdered;
+		break;
+	case 7:      // ule, ngt (less equal, unordered)
+		irOp = IROp::FCmpLessEqualUnordered;
+		break;
+	default:
+		Comp_Generic(op);
+		return;
+	}
+	ir.Write(irOp, fs, ft);
+}
+
+void IRJit::Comp_FPU2op(MIPSOpcode op) {
+	CONDITIONAL_DISABLE;
+
+	int fs = _FS;
+	int fd = _FD;
+
+	switch (op & 0x3f) {
+	case 4:	//F(fd)	   = sqrtf(F(fs));            break; //sqrt
+		ir.Write(IROp::FSqrt, fd, fs);
+		break;
+	case 5:	//F(fd)    = fabsf(F(fs));            break; //abs
+		ir.Write(IROp::FAbs, fd, fs);
+		break;
+	case 6:	//F(fd)	   = F(fs);                   break; //mov
+		ir.Write(IROp::FMov, fd, fs);
+		break;
+	case 7:	//F(fd)	   = -F(fs);                  break; //neg
+		ir.Write(IROp::FNeg, fd, fs);
+		break;
+
+	case 12: //FsI(fd) = (int)floorf(F(fs)+0.5f); break; //round.w.s
+	{
+		ir.Write(IROp::FRound, fd, fs);
+		break;
+	}
+
+	case 13: //FsI(fd) = Rto0(F(fs)));            break; //trunc.w.s
+	{
+		ir.Write(IROp::FTrunc, fd, fs);
+		break;
+	}
+
+	case 14://FsI(fd) = (int)ceilf (F(fs));      break; //ceil.w.s
+	{
+		ir.Write(IROp::FCeil, fd, fs);
+		break;
+	}
+	case 15: //FsI(fd) = (int)floorf(F(fs));      break; //floor.w.s
+	{
+		ir.Write(IROp::FFloor, fd, fs);
+		break;
+	}
+
+	case 32: //F(fd)   = (float)FsI(fs);          break; //cvt.s.w
+		ir.Write(IROp::FCvtSW, fd, fs);
+		break;
+
+	case 36: //FsI(fd) = (int)  F(fs);            break; //cvt.w.s
+		ir.Write(IROp::FCvtWS, fd, fs);
+		break;
+
+	default:
+		DISABLE;
+	}
+}
+
+void IRJit::Comp_mxc1(MIPSOpcode op)
+{
+	CONDITIONAL_DISABLE;
+
+	int fs = _FS;
+	MIPSGPReg rt = _RT;
+
+	switch ((op >> 21) & 0x1f) {
+	case 0: // R(rt) = FI(fs); break; //mfc1
+		if (rt == MIPS_REG_ZERO) {
+			return;
+		}
+		ir.Write(IROp::FMovToGPR, rt, fs);
+		return;
+
+	case 2: //cfc1
+		if (rt == MIPS_REG_ZERO) {
+			return;
+		}
+		if (fs == 31) {
+			DISABLE;
```

Comment:
Probably should just give this its own dedicated IR op, right?

I guess if we don't disable here, though, we *have* to interpret.

-[Unknown]